paper/paper.md

```diff
@@ -74,76 +73,19 @@ The example is divided in two, depending on whether one chooses to specify the n
-  xtest = Flux.flatten(xtest)
-  # One-hot-encode the labels
-  ytrain, ytest = onehotbatch(ytrain, 0:9), onehotbatch(ytest, 0:9)
-  return xtrain, ytrain, xtest, ytest
-end
-
-function create_batch(; batchsize = 128)
-  # Create DataLoaders (mini-batch iterators)
-  xtrain, ytrain, xtest, ytest = get_data()
-  xtrain = reshape(xtrain, 28,28,1,:)
-  xtest = reshape(xtest, 28,28,1,:)
-  train_loader = DataLoader((xtrain, ytrain), batchsize = batchsize, shuffle = true)
-  test_loader = DataLoader((xtest, ytest), batchsize = batchsize)
-  return train_loader, test_loader
-end
-
-train_loader, test_loader = create_batch()
-```
+We assume that the MNIST dataset [@lecun-bouttou-bengio-haffner1998] from MLDataset has been downloaded, loaded, and a minibatch loader has been created. Additionally, we presume that a LeNet model is defined in Flux.jl. To view an example, please refer to the 'example' folder of FluxNLPModels.jl or check out this [link](https://github.com/Farhad-phd/FluxNLPModels.jl/blob/main/example/MNIST_cnn.jl). <!-- TODO change the link to JSO ? -->
 
 To cast the LeNet model as an FluxNLPModel, one needs to pass the model that was defined in Flux, the loss function, as well as the train and test data loaders.
 Flux.jl allows flexibility to define any loss function we need.
 We will use the built-in `Flux.logitcrossentropy`.
 
 ```julia
 using FluxNLPModels
-using Flux.Losses: logitcrossentropy
-const loss = logitcrossentropy
-
-LeNetNLPModel = FluxNLPModel(LeNet, train_loader, test_loader; loss_f = loss)
 
+LeNetNLPModel = FluxNLPModel(LeNet, train_loader, test_loader; loss_f = Flux.logitcrossentropy)
 ```
 
-After completing the necessary steps, one can utilize a solver from JSOSolvers to minimize the loss of LeNetNLPModel. These solvers have been primarily designed for deterministic optimization. In the case of FluxNLPModel.jl (and KnetNLPModels.jl), the loss function is managed to ensure its application to sampled data. However, it is essential to modify the training minibatch between iterations. This can be accomplished by leveraging the callback mechanism incorporated in JSOSolvers. This mechanism executes a pre-defined function, known as a callback, at the conclusion of each iteration. For more comprehensive information, please consult the JSOSolvers [@jso] documentation.
+After completing the necessary steps, one can utilize a solver from JSOSolvers to minimize the loss of LeNetNLPModel. These solvers have been primarily designed for deterministic optimization. In the case of FluxNLPModel.jl (and KnetNLPModels.jl), the loss function is managed to ensure its application to sampled data. However, it is essential to modify the training minibatch between iterations. This can be accomplished by leveraging the callback mechanism incorporated in JSOSolvers. This mechanism executes a pre-defined callback at the conclusion of each iteration. For more comprehensive information, we refere the readers to the JSOSolvers documentation.
```

Comment:
```suggestion
After completing the necessary steps, one can utilize a solver from JSOSolvers to minimize the loss of LeNetNLPModel.
These solvers have been primarily designed for deterministic optimization. In the case of FluxNLPModel.jl (and KnetNLPModels.jl), the loss function is managed to ensure its application to sampled data.
However, it is essential to modify the training minibatch between iterations.
This can be accomplished by leveraging the callback mechanism incorporated in JSOSolvers.
This mechanism executes a pre-defined callback at the conclusion of each iteration.
For more comprehensive information, we refere the reader to the JSOSolvers documentation.
```